tests/integration/security/citadel_root_cert_auto_upgrade/file_mount_mtls_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package citadel_root_cert_auto_upgrade
+
+import (
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/tests/integration/security/util"
+	"istio.io/istio/tests/integration/security/util/connection"
+)
+
+func TestMtlsWithRootCertUpgrade(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+
+			istioCfg := istio.DefaultConfigOrFail(t, ctx)
+
+			namespace.ClaimOrFail(t, ctx, istioCfg.SystemNamespace)
+			ns := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "sds-citadel-flow",
+				Inject: true,
+			})
+
+			var a, b, c echo.Instance
+			echoboot.NewBuilderOrFail(t, ctx).
+				With(&a, util.EchoConfig("a", ns, false, nil, g, p)).
+				With(&b, util.EchoConfig("b", ns, false, nil, g, p)).
+				With(&c, util.EchoConfig("c", ns, false, nil, g, p)).
+				BuildOrFail(t)
+
+			checkers := []connection.Checker{
+				{
+					From: a,
+					Options: echo.CallOptions{
+						Target:   b,
+						PortName: "http",
+						Scheme:   scheme.HTTP,
+					},
+					ExpectSuccess: true,
+				},
+				{
+					From: a,
+					Options: echo.CallOptions{
+						Target:   c,
+						PortName: "http",
+						Scheme:   scheme.HTTP,
+					},
+					ExpectSuccess: true,
+				},
+				{
+					From: b,
+					Options: echo.CallOptions{
+						Target:   c,
+						PortName: "http",
+						Scheme:   scheme.HTTP,
+					},
+					ExpectSuccess: true,
+				},
+			}
+			for i := 0; i < 3; i++ {
+				for _, checker := range checkers {
+					retry.UntilSuccessOrFail(t, checker.Check, retry.Delay(time.Second), retry.Timeout(10*time.Second))
+				}
+				// Wait for at least one root upgrade to let workloads use new CA cert
+				// to set up mTLS connections.
+				time.Sleep(90 * time.Second)
```

Comment:
This test case will always wait 270 seconds and the other one in TestCitadelRootCertUpgrade will always wait for 180 seconds, together it's at least 340 seconds added to the total test.

I'm worried this will make the security e2e tests much more slower and the total time could easily pass 1 hour level. It's already slow and this is making it more painful for every single PR in Istio. (I think a test will be killed if it takes too long to complete so it could become more flaky just because it takes too much time to finish, cc @howardjohn )

Could you try to improve the test not to wait blindly for so long? It's even better to split the current security tests to two separate prow job so that they can run in parallel and saves half of the time.